### ISSUE TITLE
Having tasks with dependencies cause no tasks to show

### ISSUE THREAD

**Issue by sherlock5512** [kdheepak/taskwarrior-tui#23]
## Brief Description:
Any time I add a dependency to a task the next time I open taskwarrior-tui none of my tasks show.
The output of `task` shows the tasks still exist:

### Steps to Reproduce:
Create a Task (TASK 1)
<details>
<summary>Image:</summary>

![image](https://user-images.githubusercontent.com/22729705/97063827-e6b91180-1599-11eb-8302-f218935c2a67.png)
</details>
2. Create a Second Task (TASK 2)
<details>
<summary>Image:</summary>

![image](https://user-images.githubusercontent.com/22729705/97063855-08b29400-159a-11eb-81ea-74bfe64bd539.png)
</details>
3. Edit TASK 1 making TASK 2 it's dependency
<details>
<summary>Image:</summary>

![image](https://user-images.githubusercontent.com/22729705/97063903-53cca700-159a-11eb-8a8e-47abcf4cdcad.png)
</details>
4. Relaunch taskwarrior-cli
<details>
<summary>Image</summary>

![image](https://user-images.githubusercontent.com/22729705/97064213-34367e00-159c-11eb-8709-0c07607d7063.png)
</details>

from this point onward no tasks appear in taskwarrior-cli.

Output from `task`:
<details>
<summary>Image:</summary>

![image](https://user-images.githubusercontent.com/22729705/97064246-7233a200-159c-11eb-978f-894ceacb8bb3.png)
</details>

**Comment by kdheepak** [comment]
Thanks for opening the issue with such detailed messages. 

I tried this just now, and it works as expected for me.

![Screen Shot 2020-10-23 at 7 10 40 PM](https://user-images.githubusercontent.com/1813121/97064506-7abbb680-1563-11eb-9d46-7ce92196c41f.png)

**Comment by kdheepak** [comment]
Can also you tell me which version of `taskwarrior-tui` you are using?

**Comment by sherlock5512** [comment]
Currently I am running version 0.8.1
I'll try the latest release and see if the issue still persists.

**Comment by sherlock5512** [comment]
Just did a git pull and release build. 
the bug seems to still be there.

I have also included the output of inxi just in case the issue is something specific to my distro

![image](https://user-images.githubusercontent.com/22729705/97077827-c2cbef00-15de-11eb-83d1-886b6ca24a7a.png)

**Comment by kdheepak** [comment]
Can you share your `~/.taskrc` file with me?

**Comment by sherlock5512** [comment]
It's basically stock. 
```
# [Created by task 2.5.1 10/24/2020 00:08:51]
# Taskwarrior program configuration file.
# For more documentation, see http://taskwarrior.org or try 'man task', 'man task-color',
# 'man task-sync' or 'man taskrc'

# Here is an example of entries that use the default, override and blank values
#   variable=foo   -- By specifying a value, this overrides the default
#   variable=      -- By specifying no value, this means no default
#   #variable=foo  -- By commenting out the line, or deleting it, this uses the default

# Use the command 'task show' to see all defaults and overrides

# Files
data.location=~/.task


# Color theme (uncomment one to use)
#include /usr/share/doc/task/rc/light-16.theme
#include /usr/share/doc/task/rc/light-256.theme
#include /usr/share/doc/task/rc/dark-16.theme
#include /usr/share/doc/task/rc/dark-256.theme
#include /usr/share/doc/task/rc/dark-red-256.theme
#include /usr/share/doc/task/rc/dark-green-256.theme
#include /usr/share/doc/task/rc/dark-blue-256.theme
#include /usr/share/doc/task/rc/dark-violets-256.theme
#include /usr/share/doc/task/rc/dark-yellow-green.theme
#include /usr/share/doc/task/rc/dark-gray-256.theme
#include /usr/share/doc/task/rc/dark-gray-blue-256.theme
#include /usr/share/doc/task/rc/solarized-dark-256.theme
#include /usr/share/doc/task/rc/solarized-light-256.theme
#include /usr/share/doc/task/rc/no-color.theme
```

**Comment by kdheepak** [comment]
That is so odd. It works fine for me with a stock `~/.taskrc`. Can you run `task export` and share the results?

**Comment by kdheepak** [comment]
Can you try this version as well? In https://github.com/kdheepak/taskwarrior-tui/releases/tag/v0.8.6, all errors are bubbled up to the user now.

**Comment by sherlock5512** [comment]
Ouput of `task export` 
```
╰┤task export 
[ 
{"id":1,"depends":"faa5b76b-26f5-4488-a1d5-dc0451e9d0a5","description":"TASK 1","entry":"20201024T004013Z","modified":"20201024T005507Z","status":"pending","uuid":"b79adfc3-f7b6-454a-8b28-2a163459488b","urgency":-5}, 
{"id":2,"description":"TASK 2","entry":"20201024T004105Z","modified":"20201024T004105Z","status":"pending","uuid":"faa5b76b-26f5-4488-a1d5-dc0451e9d0a5","urgency":8}, 
{"id":0,"description":"tests","end":"20201024T084807Z","entry":"20201024T010139Z","modified":"20201024T084808Z","status":"deleted","uuid":"c4bc86d9-b6ed-42d0-8243-757f03fcfcf2","urgency":0}, 
{"id":0,"description":"pooooop","end":"20201024T084810Z","entry":"20201024T010144Z","modified":"20201024T084812Z","status":"deleted","uuid":"b6452b89-5cea-46cc-82b8-381d502417b6","urgency":0} 
] 
```
NOTE: the deleted tasks were just me checking that there wasn't anything going wrong with taskwarrior itself

version 0.8.6 immediately crashes with the following error
``` 
╰┤taskwarrior-tui 
[taskwarrior-tui error]: Failed to create a Task from JSON 
```

**Comment by kdheepak** [comment]
And what version of taskwarrior are you using? Can you run `task --version`?

**Comment by sherlock5512** [comment]
```
╰┤task --version 
2.5.1 
```

**Comment by kdheepak** [comment]
I think I found the issue! For some reason, your task urgency is set to a negative value `-5` when it depends on another task. And, the dependency I'm using assumes that task the urgency is always a string, floating point or unsigned integer.

```
    /// UDA is an integer
    U64(u64),
```

https://github.com/matthiasbeyer/task-hookrs/blob/2953cf02b3c56b164a44e95fac5aae6f38954620/src/uda.rs#L22-L23

In any case, the quick hacky fix for this is to use the following in your `.taskrc`. 

```bash
urgency.waiting.coefficient                 0.0 # waiting task
urgency.blocked.coefficient                 0.0 # blocked by other tasks
```

I'll open an issue on [`task-hookrs`](https://github.com/matthiasbeyer/task-hookrs), so that we can support more kinds of UDAValues.

**Comment by sherlock5512** [comment]
Thanks, the Hacky fix is good for the moment.
And while I don't really understand rust it seems like that is likely the root cause.

**Comment by kdheepak** [comment]
You can also use this for now and it should work fine:

```plaintext
urgency.waiting.coefficient                 -3.0001 # waiting task
urgency.blocked.coefficient                 -5.0001 # blocked by other tasks
```

It's just negative integers that bork. I'm also a little new to rust :) so I'm still figuring out the order of the deserialization. But hopefully this will be a quick and easy fix. I'll leave this issue open till this is addressed properly.

**Comment by kdheepak** [comment]
Can you try the latest release? I believe this issue is resolved in this release: https://github.com/kdheepak/taskwarrior-tui/releases/tag/v0.8.9

**Comment by sherlock5512** [comment]
I can confirm that the issue is now resolved when using the same test data as before.

**Comment by kdheepak** [comment]
Thanks!
